### ISSUE TITLE
AttributeError: 'str' object has no attribute 'reprcrash' while using pytest.mark.xfail

### ISSUE THREAD

**Issue by metheuspsc** [pytest-dev/pytest-rerunfailures#239]
Hello!

In my project,  I use @pytest.mark.flaky(reruns=3, rerun_except=CustomException) using a specific custom Exception on a couple of tests. That causes the following error that looks similar to issue #175:

> INTERNALERROR> E               File "/Users/matheuspessoa/PycharmProjects/bigcommerce-adapter/venv/lib/python3.10/site-packages/pytest_rerunfailures.py", line 331, in _matches_any_rerun_except_error
> INTERNALERROR> E                 if re.search(rerun_regex, report.longrepr.reprcrash.message):
> INTERNALERROR> E             AttributeError: 'str' object has no attribute 'reprcrash'
> INTERNALERROR> E           assert False

 So I suggest the same fix, which I tested locally and worked.

 
 ```def _matches_any_rerun_except_error(rerun_except_errors, report):
    for rerun_regex in rerun_except_errors:
        try:
            if re.search(rerun_regex, report.longrepr.reprcrash.message):
                return True
        except AttributeError:
            if re.search(rerun_regex, report.longreprtext):
                return True
    return False```python

**Comment by danylokholodov** [comment]
@metheuspsc Thank you for your solution! It helped.

**Comment by icemac** [comment]
@metheuspsc Could you please create a pull request for your fix?

**Comment by metheuspsc** [comment]
> @metheuspsc Could you please create a pull request for your fix?

Done

**Comment by icemac** [comment]
Fixed by #240.
